include/exadg/postprocessor/hdf5_generator_base.h

```diff
@@ -0,0 +1,149 @@
+{
+  using point_value_type = typename dealii::Point<dim>::value_type;
+
+  HDF5DataBase();
+
+  bool        write_output;
+  std::string directory;
+  std::string filename;
+
+  bool update_points_before_evaluation;
+
+  double start_time;
+  double end_time;
+  double interval_time;
+
+  std::vector<dealii::Point<dim>> evaluation_points;
+
+  void
+  print(dealii::ConditionalOStream & pcout) const;
+};
+
+template<int dim, typename Number>
+class HDF5GeneratorBase
+{
+public:
+  using VectorType = dealii::LinearAlgebra::distributed::Vector<Number>;
+
+  HDF5GeneratorBase(MPI_Comm const & comm);
+  virtual ~HDF5GeneratorBase() = default;
+
+
+  void
+  setup_base(dealii::Triangulation<dim> const & tria_in,
+             dealii::Mapping<dim> const &       mapping_in,
+             HDF5DataBase<dim> const &          hdf5_data_in);
+
+  void
+  evaluate(VectorType const & solution, double const & time, int const & time_step_number);
+
+  void
+  add_scalar_quantity_dataset(std::string const & name);
+
+  void
+  add_vector_quantity_datasets(std::string const & name);
+
+  void
+  write_scalar_quantity(std::string const & name, dealii::Vector<Number> const & hyperslab_data);
+
+  template<int n_components>
+  void
+  extract_component(dealii::Vector<Number> &                                     dst,
+                    std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                    unsigned int const                                           comp) const
+  {
+    AssertIndexRange(comp, n_components);
+
+    auto iter = dst.begin();
+    for(auto const & val : values)
+    {
+      *iter = val[comp];
+      ++iter;
+    }
+  }
+
+  dealii::Utilities::MPI::RemotePointEvaluation<dim> const &
+  get_remote_evaluator() const;
+
+private:
+  using point_value_type = typename HDF5DataBase<dim>::point_value_type;
+
+  virtual void
+  do_evaluate(VectorType const & solution) = 0;
+
+  void
+  setup_remote_evaluator();
+
+  void
+  reinit_remote_evaluator();
+
+  void
+  create_hdf5_file();
+
+  void
+  add_evaluation_points_dataset(dealii::HDF5::Group & group, std::string const & name);
+
+  void
+  fill_evaluation_points(std::string const & name);
+
+  void
+  add_time_dataset(dealii::HDF5::Group & group, std::string const & name);
+
+  void
+  write_time(double time);
+
+  MPI_Comm const                                                      mpi_comm;
+  dealii::SmartPointer<dealii::Triangulation<dim> const>              tria;
+  dealii::SmartPointer<dealii::Mapping<dim> const>                    mapping;
+  std::shared_ptr<dealii::Utilities::MPI::RemotePointEvaluation<dim>> remote_evaluator;
+  HDF5DataBase<dim>                                                   hdf5_data;
+  unsigned int                                                        counter = 0;
+  unsigned int                                                        n_out_samples;
+  bool                                                                reset_counter = true;
```

Comment:
initialization should be done in constructor in my opinion (is done like this almost everywhere in ExaDG).

Reply:
resolved in [5f34cc0](https://github.com/exadg/exadg/pull/255/commits/5f34cc07efbf9106a7441fbd8460f044066469ed)